README.md

```diff
@@ -31,77 +31,101 @@ A Prometheus exporter for [WireGuard](https://www.wireguard.com), written in Rus
+### Pre-built binaries
 
-Alternatively, as long as you have Wireguard on your host kernel with some Wireguard interfaces running, you can use Docker. For example:
+Coming soon, subcribe to [#59](https://github.com/MindFlavor/prometheus_wireguard_exporter/issues/59)
 
-```sh
-docker run -it --rm --init --net=host --cap-add=NET_ADMIN mindflavor/prometheus-wireguard-exporter
-# Check it's up
-docker run -it --rm alpine:3.12 wget -qO- http://localhost:9586/metrics
-```
+### Docker
+
+1. You need Docker installed
+1. You need [WireGuard](https://www.wireguard.com) installed in your host kernel
+1. You need some Wireguard interfaces running
+1. Download and run the container with:
 
-ℹ️ The Docker image is compatible with `amd64`, `386`, `arm64`, `armv7` and `armv6` CPUs.
+    ```sh
+    docker run -d --init --net=host --cap-add=NET_ADMIN --name wgexporter mindflavor/prometheus-wireguard-exporter
+    ```
 
-To update the image, you can use
+1. Check it's up by visiting [http://localhost:9586/metrics](http://localhost:9586/metrics)
+
+You can then update the image with
 
 ```sh
 docker pull mindflavor/prometheus_wireguard_exporter
 ```
 
-You can also build it with:
+Or use a [tagged image](https://hub.docker.com/r/mindflavor/prometheus-wireguard-exporter/tags) such as `:3.5.1`.
+
+You can also build the Docker image from source (you need `git`) with:
 
 ```sh
-docker build -t mindflavor/prometheus_wireguard_exporter https://github.com/MindFlavor/prometheus_wireguard_exporter.git
+docker build -t mindflavor/prometheus_wireguard_exporter https://github.com/MindFlavor/prometheus_wireguard_exporter.git#master
 ```
 
-## Compilation
+### Build from source
 
-To compile the latest master version:
+1. You need [Rust](https://www.rust-lang.org/tools/install) installed
+1. You need [WireGuard](https://www.wireguard.com) installed on your host
+1. You need `wg` accessible in your path. The tool will call `wg show <interface(s)>|all dump` and of course will fail if the `wg` executable is not found.
+1. You need some Wireguard interfaces running
+1. You need `git` installed
+1. Clone the repository with
 
-```bash
-git clone https://github.com/MindFlavor/prometheus_wireguard_exporter.git
-cd prometheus_wireguard_exporter
-cargo install --path .
-```
+    ```sh
+    git clone https://github.com/MindFlavor/prometheus_wireguard_exporter.git
+    cd prometheus_wireguard_exporter
+    ```
 
-If you want the latest release you can simply use:
+1. Compile the program with
 
-```bash
-cargo install prometheus_wireguard_exporter
-```
+    ```sh
+    cargo install --path .
+    ```
 
-You can also build the Docker image with
+    💁 If you encounter errors, please try:
+    * Update your Rust installation
 
-```sh
-docker build -t mindflavor/prometheus-wireguard-exporter .
-```
+        ```sh
+        rustup update
+        cargo install -path .
+        ```
+
+    * Use the same Rust version I used: `rustc 1.55.0-nightly (798baebde 2021-07-02)`
```

Comment:
@MindFlavor would you have exact `rustup` instructions to do this? I can't figure where to find the corresponding `rustup install <string>` somehow 😢

Reply:
Theoretically any 2018-compliant rustc compiler should be able to compile the code, even if older compared to the version I'm using (I should probably test it with a stable release as well 🤷‍♂️). 

That said, the wording I've used: "*Use the same Rust version I used*" is wrong.

We could rephrase it as "*the code should compile with any relatively recent, 2018-compliant rustc version. As a frame of reference, the last release was tested on `rustc 123 xyz`*". Do you think it would be better?

Reply:
👍

Actually I realized it's built with the rust Docker image `:1-bullseye-slim` which runs with `rustc 1.53.0 (53cb7b09b 2021-06-17)`, I'll put that one as reference if that's ok with you, as it doesn't sound like we require to run on 1.55.0-nightly (which might be more tedious to setup for rust beginners).